frontend/lib/resolution/Resolver.cpp

```diff
@@ -5706,10 +5714,15 @@ void Resolver::exit(const Dot* dot) {
                        /* isParenless */ true, actuals);
     auto inScope = currentScope();
     auto inScopes = CallScopeInfo::forNormalCall(inScope, poiScope);
-    auto rr = resolveGeneratedCall(dot, &ci, &inScopes, name.c_str());
-
-    auto baseDomainType = rr.result.exprType().type()->toDomainType();
-    r.setType(QualifiedType(QualifiedType::CONST_VAR, baseDomainType));
+    auto resultType = QualifiedType();
+    if (!shouldSkipCallResolution(this, dot, actualAsts, ci)) {
+      auto rr = resolveGeneratedCall(dot, &ci, &inScopes, name.c_str());
+      if (!rr.result.mostSpecific().isEmpty()) {
+        auto baseDomainType = rr.result.exprType().type()->toDomainType();
```

Comment:
`exprType` could be `null` even if `mostSpecific` is not empty. However, why can't you just set `resultType = rr.result.exprType()`? Or, better yet, do `rr.noteResult(..)`, which will also do the work of propagating `compilerError`, tracking uses of PoI functions, etc.?

Reply:
There is a `noteResultWithoutError` as well, I think.